docs/how-to-work-on-coding-challenges.md

```diff
@@ -31,96 +31,135 @@ Before you work on the curriculum, you would need to set up some tooling to help
 
 ## Challenge Template
 
-Below is a template of what the challenge markdown files look like currently.  To see the streamlined template we will be adopting see [below](#upcoming-challenge-template).
-
 ````md
+
 ---
 id: Unique identifier (alphanumerical, MongoDB_id)
-title: Challenge Title
-challengeType: 0
+title: 'Challenge Title'
+challengeType: Integer, defined in `client/utils/challengeTypes.js`
 videoUrl: 'url of video explanation'
+forumTopicId: 12345
 ---
 
-## Description
+# --description--
 
-<section id='description'>
-A Description of the challenge and what is required to pass
-</section>
+Challenge description text, in markdown
 
-## Instructions
+```html
+<div>
+  example code
+</div>
+```
 
-<section id='instructions'>
-Instructions about what exactly needs to be done.
-</section>
+# --instructions--
 
-## Tests
+Challenge instruction text, in markdown
 
-<section id='tests'>
+# --hints--
 
-```yml
-tests:
-  - text: Should return "foo"
-    testString: 'A stringified function possibly using Chai asserts'
+Tests to run against user code, in pairs of markdown text and codeblock test code.
+
+```js
+Code for test one
 ```
 
-</section>
+More instructions in markdown syntax
 
-## Challenge Seed
+```js
+More code
+```
 
-<section id='challengeSeed'>
+# --seed--
+
+## --before-user-code--
 
-<div id='{ext}-seed'>
+```lang
+Code evaluated before the user’s code.
+```
 
-```{ext}
-Code displayed in the editor by default.
+## --after-user-code--
 
-This is a required section for the challenge.
+```lang
+Code evaluated after the user’s code, and just before the tests
 ```
 
-</div>
+## --seed-contents--
+
+Boilerplate code to render to the editor
```

Comment:
```suggestion
Boilerplate code to render to the editor. This section should only contain code inside backticks, like the following:
```
At first glance people could assume that this text is okay (since it's in the template). However, if someone puts some raw text here, the parser will throw.